test/spec/proxySpec.js

```diff
@@ -1148,6 +1146,86 @@ describe('proxy', function() {
       // }, 100);
     });
   });
+  describe('Already proxified values', function() {
+    it("shouldn't re-proxify (it should deep clone) proxified values from different instances", function() {
```

Comment:
No. Having the same source object isn't an issue because we deep clone before we proxify. This is test the case for:
```js
ProxyA.newValue= ProxyB.proxifiedObject
```
This test makes sure that `ProxyA.newValue.addOp = 1` doesn't trigger ProxyB.